lib/zclient.c

```diff
@@ -2519,7 +2519,7 @@ static bool zapi_nexthop_update_decode(struct stream *s, struct prefix *match,
 	STREAM_GETW(s, nhr->nexthop_num);
 
 	for (i = 0; i < nhr->nexthop_num; i++) {
-		if (zapi_nexthop_decode(s, &(nhr->nexthops[i]), 0, 0) != 0)
+		if (zapi_nexthop_decode(s, &(nhr->nexthops[i]), 0, nhr->message) != 0)
 			return false;
```

Comment:
it looks like a bug. a separate commit should be put in place along with a test that demonstrates the issue.